chemprop/train/make_predictions.py

```diff
@@ -241,14 +243,25 @@ def predict_and_save(args: PredictArgs, train_args: TrainArgs, test_data: Molecu
         for datapoint in full_data:
             writer.writerow(datapoint.row)
 
+    # Return predicted values
     avg_preds = avg_preds.tolist()
     
-    return avg_preds
+    if return_invalid_smiles:
+        full_preds = []
+        for full_index in range(len(full_data)):
+            valid_index = full_to_valid_indices.get(full_index, None)
+            preds = avg_preds[valid_index] if valid_index is not None else ['Invalid SMILES'] * num_tasks
```

Comment:
Is it possible to also return the invalid SMILES themselves, or is it not really beneficial?

Reply:
I'm not sure that it's beneficial. But if people do want it, the right place is in the load_data function where these are getting split up in the first place. I've added it there as one of the outputs (unused in the full workflow). Is that a reasonable return structure do you think?

Reply:
@alongd I've actually rethought this and I don't want to add another output to load_data where it would break existing code for people. The right place would be in a separate data/utils function. I've added new functions for that from both files and lists.